### PR TITLE
Test jwt signing key

### PR DESCRIPTION
**Pull request by cseed**
stacked on: https://github.com/hail-is/hail/pull/6139

create test jwt-secret-key secret and corresponding Hail service accounts.

### CODE REVIEW

File: build.yaml [hunk 1]
```diff
@@ -31,14 +26,34 @@ steps:
...
     - default_ns
     - deploy_batch_sa
+ - kind: runImage
+   name: create_accounts
+   image: "{{ base_image.image }}"
+   script: |
+     set -e
+     echo creating jwt-signing-key secret...
+     python3 -c "
+     import secrets, json, jwt
```
Comment (jigold) [inline build.yaml]: need to not do this in production

File: batch/deployment.yaml [hunk 2]
```diff
@@ -42,13 +42,10 @@ spec:
...
-         - name: jwt-secret
-           mountPath: /jwt-secret
+         - name: jwt-secret-key
+           mountPath: /jwt-secret-key
            readOnly: true
-         - name: gsa-key
-           mountPath: /batch-gsa-key
-           readOnly: true
-         - name: jwt
+         - name: batch-jwt
```
Comment (jigold) [inline batch/deployment.yaml]: What happened to the gsa-key? Did you remove it because it's not used?

File: build.yaml [hunk 3]
```diff
@@ -533,7 +548,7 @@ steps:
    name: test_pipeline
    namespace:
      valueFrom: batch_pods_ns.name
-   config: pipeline/test-pipeline-pod.yaml
+   config: pipeline/test-pipeline-pod.yaml
```
Comment (jigold) [inline build.yaml]: whitespace can be deleted.

File: build.yaml [hunk 4]
```diff
@@ -31,14 +26,34 @@ steps:
...
+       f.write(secret_key)
+     payload = {'ksa_name':'unused','gsa_email':'hail-test@hail-vdc.iam.gserviceaccount.com','bucket_name':'hail-test','gsa_key_secret_name':'test-gsa-key'}
+     with open('jwt', 'w') as f:
+       f.write(jwt.encode(json.dumps(payload), secret_key))
+     "
+     kubectl -n {{ default_ns.name }} create secret generic jwt-secret-key --from-file=./secret-key
+     kubectl -n {{ default_ns.name }} create secret generic batch-jwt --from-file=./jwt
+     kubectl -n {{ default_ns.name }} create secret generic ci2-jwt --from-file=./jwt
+     kubectl -n {{ batch_pods_ns.name }} create secret generic test-jwt --from-file=./jwt
+   serviceAccount: ci2-agent
```
Comment (jigold) [inline build.yaml]: Why is this necessary? Where does the gsa_key come from for this service account?

File: build.yaml (hunk 1 above)
Comment (cseed) [reply build.yaml]: I added a `scopes` field to step which is a list of scopes to run in.  There are currently two scopes, test and deploy.  If no scopes are specified, the step is always run.  I made this step scopes: ["test"].

File: batch/deployment.yaml (hunk 2 above)
Comment (cseed) [reply batch/deployment.yaml]: I incorrectly deleted it. I put it back.  If we're deploy, I use the original key, otherwise I use test-gsa-key.

File: build.yaml (hunk 4 above)
Comment (cseed) [reply build.yaml]: This is not a Hail service account, it is a k8s service account.  It has no gsa key.  It has the permissions necessary to run the kubectl create secret commands above in both the default and batch_pods namespaces.

File: ci2/ci/build.py [hunk 5]
```diff
@@ -686,20 +708,20 @@ def config(self, deploy):  # pylint: disable=unused-argument
 
 
 class CreateDatabaseStep(Step):
-    def __init__(self, code, deploy, name, deps, database_name, namespace):
-        super().__init__(name, deps)
+    def __init__(self, params, database_name, namespace):
+        super().__init__(json)
```
Comment (jigold) [inline ci2/ci/build.py]: I think this should be params

**Comment by cseed** [comment]
I undid the change which switched reading the jwt-secret-key as binary.  That will break all the existing stuff, although when we rotate the keys, we should make that change so our secret key is sampled from the full space of keys.

File: batch/test-batch-pod.yaml [hunk 6]
```diff
@@ -12,15 +12,15 @@ spec:
          fieldRef:
            fieldPath: status.podIP
      - name: HAIL_TOKEN_FILE
-       value: '/jwt/jwt'
+       value: '/test-jwt/jwt'
      - name: BATCH_URL
        value: http://batch.{{ default_ns.name }}
     volumeMounts:
-      - mountPath: /jwt
+      - mountPath: /test/jwt
```
Comment (jigold) [inline batch/test-batch-pod.yaml]: Should this path be the same as the path above?

File: ci2/ci/build.py (hunk 5 above)
Comment (cseed) [reply ci2/ci/build.py]: Good catch, thanks, fixed.  I guess lint didn't catch it because we're importing the json package.

File: build.yaml [hunk 7]
```diff
@@ -31,11 +26,7 @@ steps:
        valueFrom: default_ns.name
    public: false
    secrets:
-    # hail|test-batch
-    - user-jwt-vkqfw
-    - gsa-key-2x975
-    # pipeline|test-user
-    - pipeline-test-0-1--hail-is-service-account-key
```
Comment (jigold) [inline build.yaml]: I think deleting the pipeline service account will fail until my PR goes in. Trying to debug that still.
